code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+#sum(datafile$obds_in_month) check this matches publication when testing prevous data.
+
+#strip out code 100s
+
+
+datafile<-datafile %>% filter(datafile$reas1!="Code 100")
+#table(datafile$reas1) # check code 100 has been removed
+
+#Create new variables
+
+datafile<-datafile %>% mutate(areaname=" ")
+datafile<-datafile %>% mutate(year="2019-20")
+
+### 4a. Rename variables ----
+#( may not be needed once full RAP programs used as variables will have been renamed in earlier process)
+# datafile <- datafile %>%
+# rename(nhs_board=Healthboard,
+#          chi_number=CHINo,
+#          age_at_rdd=AGEATRDD,
+#          postcode=PatientPostcode,
+#          local_authority_code=LocalAuthorityArea,
+#          date_of_birth=PatientDOB,
+#          discharge_specialty_nat_code=SpecialtyCode,
+#          date_referred_for_sw_assessment=DateReferralReceived,
+#          date_declared_medically_fit=Readyfordischargedate,
+#          REASONFORDELAY=REASONFORDELAY,
+#          dd_code_2=REASONFORDELAYSECONDARY,
+#          out_of_area_case_indicator=Outofareacaseindicator,
+#          admission_date=OriginalAdmissionDate,
+#          sex_code=Gender,
+#          discharge_date=DateDischarge,
+#          discharge_to_code=DischargeReason,
+#          census_flag=CENSUSFLAG,
+#          delay_description=DELAY_DESCRIPTION)
+
+#Create age_grpgGrouping
+
+datafile<-datafile%>% mutate(age_grp=
+                               if_else(AGEATRDD<75, "18-74",
+                                       if_else(AGEATRDD>=75, "75+", " ")))
+
+
+
+
+
+#save out file for main bed days file
+ 
+
+#write_sav(datafile,paste0(filepath,"main bed days file.sav")) # save out file
+
+#set function
+#'%!in%' <- function(x,y)!('%in%'(x,y))
+
+#Save census data file
+
+
+datafile2 <- filter(datafile, !REASONFORDELAYSECONDARY %in%c("26X","46X") & CENSUSFLAG=="Y")
+
+
+#write_sav(datafile2,paste0(filepath,"main census file.sav")) # save out file
+
+
+
+# 5. Create Reason 2 Groupings ----
+
+##Sub grouping reason_grp for code 9s are currently being classed as H&SC or Pat/Fam reasons. Ensure they are just code 9s.
+##Also bring in Transport as a separate category.
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION=="Adults with Incapacity Act", "Adults with Incapacity Act",reas2))
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION!="Adults with Incapacity Act" & reas1=="Code 9", "Other code 9 reasons(not AWI)",reas2))
+
+datafile3<-datafile2 %>% 
+  mutate(reas2 = as.character(reas2))%>% 
+  mutate(reas2=
+           if_else(DELAY_DESCRIPTION=="Awaiting availability of transport","Transport",reas2))
+
+
+#write_sav(datafile3,paste0(filepath,"main census file Tabs 1 3 6.sav")) # save out file
+
+### 6. Get reason / age_grp breakdown for Scotland ----
+
+datafile4<-datafile3
+datafile4$level<-"1"
+
+datafile4$areaname<-"Scotland"
```

Comment:
See "Style Guide", "Pipes", and "Code Needs Refactoring" sections in general feedback comments.

It may seem arbitrary but our best practice for adding new columns would be to use the `mutate()` function, it's easier to understand and the code is more efficient. However, this is usually used to add columns as a function of existing ones. 
```suggestion
datafile %<>% mutate(level = "1", areaname = "Scotland")
```